pkg/cli/admin/upgrade/status/status.go

```diff
@@ -0,0 +1,247 @@
+// Package status contains a command for displaying a cluster's current status.
+package status
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/blang/semver"
+	configv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+const (
+	// ClusterStatusFailing is set on the ClusterVersion status when a cluster
+	// cannot reach the desired state. It is considered more serious than Degraded
+	// and indicates the cluster is not healthy.
+	ClusterStatusFailing = configv1.ClusterStatusConditionType("Failing")
+)
+
+func NewOptions(streams genericclioptions.IOStreams) *Options {
+	return &Options{
+		IOStreams: streams,
+	}
+}
+
+func New(f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewOptions(streams)
+	cmd := &cobra.Command{
+		Use:   "status",
+		Short: "Display the cluster's current status",
+		Long: templates.LongDesc(`
+			Display the cluster's current status.
+
+			Retrieve the current version info and display whether an upgrade is in progress or
+			whether any errors might prevent an upgrade, as well as show the suggested
+			updates available to the cluster. Information about compatible updates is periodically
+			retrieved from the update server and cached on the cluster - these are updates that are
+			known to be supported as upgrades from the current version.
+		`),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+	flags := cmd.Flags()
+	flags.BoolVar(&o.IncludeNotRecommended, "include-not-recommended", o.IncludeNotRecommended, "Display additional updates which are not recommended based on your cluster configuration.")
```

Comment:
-1 for adding the `nclude-not-recommended` flag here. IMO adding the flag is a bad idea because `oc adm upgrade status` should not talk about update recommendations. It is a separate subcommand line `oc adm upgrade channel`.

Reply:
Update recommendations are a big part of `oc adm upgrade` today, and it makes sense to me to bring those along for the same read vs. write distinction [we're using to motivate the existence of a `status` subcommand][1].  "There are update recommendations" is basically "your current release lacks features and bugfixes your channel suggests  you are interested in".  I could see a new `oc adm status` that just talked about ClusterVersion `conditions` and maybe said "there are available updates" without going into further details.  But by the time you get down to `oc adm upgrade status`, talking about update options seems like a natural fit, doesn't it?

[1]: https://github.com/openshift/oc/pull/1470#issue-1759322634